src/components/modal/index.tsx

```diff
@@ -1,44 +1,51 @@
 import _ from 'lodash';
-import PropTypes from 'prop-types';
-import React from 'react';
-import {StyleSheet, Modal as RNModal, TouchableWithoutFeedback} from 'react-native';
+import React, {Component} from 'react';
+import {StyleSheet, Modal as RNModal, ModalProps, TouchableWithoutFeedback, GestureResponderEvent} from 'react-native';
 import {BlurView} from '@react-native-community/blur';
 import {Constants} from '../../helpers';
-import {BaseComponent} from '../../commons';
-import TopBar from './TopBar';
+import {asBaseComponent} from '../../commons/new';
+import TopBar, {ModalTopBarPropTypes} from './TopBar';
 import View from '../../components/view';
 
-/**
- * @description: Component that present content on top of the invoking screen
- * @extends: Modal
- * @extendslink: https://facebook.github.io/react-native/docs/modal.html
- * @gif: https://media.giphy.com/media/3oFzmfSX8KgvctI4Ks/giphy.gif
- * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/ModalScreen.js
- */
-export default class Modal extends BaseComponent {
-  static displayName = 'Modal';
-  static propTypes = {
+export {ModalTopBarPropTypes};
+export interface ModalPropTypes extends ModalProps {
```

Comment:
Let's rename to `ModalProps`
In general, I would like to unify the naming of our component typings and make it shorter. 
I know there are already some component that the typings are export as `...PropTypes`, we'll handle them later.

Reply:
Done